src/data_channel/mod.rs

```diff
@@ -81,7 +81,7 @@ impl DataChannel {
     pub async fn accept(
         association: &Arc<Association>,
         config: Config,
-        existing_channels: Vec<DataChannel>,
+        existing_channels: &[Arc<DataChannel>],
```

Comment:
It would be nice if this API was less restrictive i.e. could be use with other types than `Arc`, especially since we aren't returning `DataChannel` wrapped in an `Arc` form `new`.

A super unrestrictive version would be something like https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=efdb68f631cab7c41e9c61b33b2cf6ac